io/LasWriter.cpp

```diff
@@ -1185,45 +1206,28 @@ void LasWriter::finishOutput()
         finishLasZipOutput();
     else if (d->opts.compression == las::Compression::LazPerf)
         finishLazPerfOutput();
-
-    log()->get(LogLevel::Debug) << "Wrote " <<
-        m_summaryData->getTotalNumPoints() <<
+    log()->get(LogLevel::Debug) << "Wrote " << d->summary.getTotalNumPoints() <<
         " points to the LAS file" << std::endl;
 
-    OLeStream out(m_ostream);
-
-    // addVlr prevents any eVlrs from being added before version 1.4.
-    m_lasHeader.setEVlrOffset(m_eVlrs.size() ? (uint32_t)m_ostream->tellp() : 0);
-    for (auto vi = m_eVlrs.begin(); vi != m_eVlrs.end(); ++vi)
+    // addVlr prevents any evlrs from being added before version 1.4.
+    d->header.evlrOffset = (m_evlrs.size() ? (uint32_t)m_ostream->tellp() : 0);
+    for (const las::Evlr& evlr : m_evlrs)
     {
-        ExtLasVLR evlr = *vi;
-        out << evlr;
+        std::vector<char> buf = evlr.headerData();
+        m_ostream->write((const char *)buf.data(), buf.size());
+        m_ostream->write((const char *)evlr.data(), evlr.dataSize());
     }
 
-    // Reset the offset/scale since it may have been auto-computed
-    try
-    {
-        m_lasHeader.setScaling(m_scaling);
-    }
-    catch (const LasHeader::error& err)
-    {
-        throwError(err.what());
-    }
+    // Refill the header since the offset/scale may have been auto-computed.
+    fillHeader();
 
     // The summary is calculated as points are written.
-    try
-    {
-        m_lasHeader.setSummary(*m_summaryData);
-    }
-    catch (const LasHeader::error& err)
-    {
-        throwError(err.what());
-    }
-
-    out.seek(0);
-    out << m_lasHeader;
-    out.seek(m_lasHeader.pointOffset());
+    las::setSummary(d->header, d->summary);
 
+    std::vector<char> buf = d->header.data();
+    m_ostream->seekp(0);
+    m_ostream->write(buf.data(), buf.size());
+    m_ostream->seekp(d->header.pointOffset);  //ABELL ?
```

Comment:
I think this might be holdover from LASzip's API behavior, where we were passing in our ostream and it was moving things around and not leaving it where we expected.

Reply:
That sounds right. I'll add a comment.